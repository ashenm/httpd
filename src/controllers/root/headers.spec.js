const headers = require('./headers');

describe('root/headers.js', () => {
  const response = {
    json: jest.fn().mockReturnThis(),
    render: jest.fn().mockReturnThis(),
    status: jest.fn().mockReturnThis()
  };

  const request = {
    accepts: jest.fn(),
    headers: { 'X-Key': 'x-test-value' }
  };

  describe('text/html', () => {
    // given
    beforeAll(() => {
      request.accepts.mockReturnValueOnce(true);
    });

    // when
    beforeAll(() => {
      headers(request, response);
    });

    // then
    it('Should attempt to resolve content negotiation selection with valid arity', () => {
      expect(request.accepts.mock.calls).toMatchSnapshot();
    });

    it('Should attempt to render response view with valid arity', () => {
      expect(response.render.mock.calls).toMatchSnapshot();
    });

    afterAll(jest.clearAllMocks);
  });

  describe('text/plain', () => {
    // given
    beforeAll(() => {
      request.accepts.mockReturnValueOnce(false);
    });

    // when
    beforeAll(() => {
      headers(request, response);
    });

    // then
    it('Should dispatch response status with valid arity', () => {
      expect(response.status.mock.calls).toMatchSnapshot();
    });

    it('Should dispatch response body with valid arity', () => {
      expect(response.json.mock.calls).toMatchSnapshot();
    });

    afterAll(jest.clearAllMocks);
  });
});
