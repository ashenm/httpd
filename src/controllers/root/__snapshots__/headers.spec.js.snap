// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`root/headers.js text/html Should attempt to render response view with valid arity 1`] = `
[
  [
    "headers",
    {
      "headers": {
        "X-Key": "x-test-value",
      },
    },
  ],
]
`;

exports[`root/headers.js text/html Should attempt to resolve content negotiation selection with valid arity 1`] = `
[
  [
    "text/html",
  ],
]
`;

exports[`root/headers.js text/plain Should dispatch response body with valid arity 1`] = `
[
  [
    {
      "X-Key": "x-test-value",
    },
  ],
]
`;

exports[`root/headers.js text/plain Should dispatch response status with valid arity 1`] = `
[
  [
    200,
  ],
]
`;
